internal/auth/credentials.go

```diff
@@ -16,14 +16,24 @@ func NewCredentialsStore() (credentials.Store, error) {
 	}
 
 	opts := credentials.StoreOptions{AllowPlaintextPut: false}
-	primaryStore, err := credentials.NewStore(configPath, opts)
+	notationStore, err := credentials.NewStore(configPath, opts)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create credential store from config file: %w", err)
 	}
 
-	fallbackStore, err := credentials.NewStoreFromDocker(opts)
+	dockerStore, err := credentials.NewStoreFromDocker(opts)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create credential store from docker config file: %w", err)
 	}
-	return credentials.NewStoreWithFallbacks(primaryStore, fallbackStore), nil
+
+	if !notationStore.IsAuthConfigured() && !dockerStore.IsAuthConfigured() {
+		// if both notation config and docker config do not have auth
+		// configuration, fallback to the platform-default docker credentials
+		// helper (if available)
+		osDefaultStore, ok := credentials.NewDefaultNativeStore()
+		if ok {
+			return credentials.NewStoreWithFallbacks(notationStore, dockerStore, osDefaultStore), nil
+		}
+	}
```

Comment:
Why cant we return defualt trust store here when we know both notation's and docker's are not configured?

Reply:
Good point! I think it makes sense to return the default store here.